server/svix-server/src/core/security.rs

```diff
@@ -77,11 +128,83 @@ where
     }
 }
 
+pub struct AuthenticatedOrganization {
+    pub permissions: Permissions,
+}
+
+#[async_trait]
+impl<B> FromRequest<B> for AuthenticatedOrganization
+where
+    B: Send,
+{
+    type Rejection = Error;
+
+    async fn from_request(req: &mut RequestParts<B>) -> Result<Self> {
+        let permissions = Permissions::from_request(req).await?;
+        if permissions.type_ == KeyType::Application {
```

Comment:
```suggestion
        if permissions.type_ != KeyType::Organization {
```
What if we add a new type called `Foobar`? You've now given them org level access.